src/modules/session/thunks.ts

```diff
@@ -72,6 +76,42 @@ export const createAccount = createAsyncThunk(
   }
 );
 
+/**
+ * Reset user password. Navigate to the login page to have user
+ * enter their new password.
+ */
+export const resetPassword = createAsyncThunk(
+  "session/resetPassword",
+  async (body: ResetPasswordRequest, { dispatch }) => {
+    const resetPasswordResponse = await dispatch(
+      sessionApi.endpoints.resetPassword.initiate(body)
+    );
+
+    if ("error" in resetPasswordResponse) {
+      dispatch(
+        setToastMessage({
+          message: "An error occured while resetting your password",
+          severity: "error",
+        })
+      );
+
+      return;
+    }
+
+    dispatch(
+      setToastMessage({
+        heading: "Password changed!",
+        message: "Login with the newly defined password.",
+        severity: "success",
+      })
+    );
+
+    setTimeout(() => {
+      window.location.pathname = "login";
+    }, 2000);
+  }
```

Comment:
@scandycuz I wanted to get your thoughts on this. In the mocks it shows the redirect to login and then after, you show the toast. I tried different ways to do this but was unsuccessful, the only way that i think we can accomplish that is through sessionStorage and then dispatching the toast if content is present in sessionStorage in the login page. Do you have any other things we can try to accomplish what they want? should i go with the sessionStorage solution? or should i ask design team if they are okay with this?

Reply:
I'm also curious to see if there is another way around this without changing too much of the current structure. Would it be possible to add the `useNavigate` hook to the `Toast` component and then use dispatch to provide an optional navigation route to then display the message?

Reply:
I can leave as is for now and when we merge the other PR we can make the change here, i dont want to copy/paste code around

Reply:
Thanks, implemented this in the latest push.